std/json.d

```diff
@@ -96,11 +96,72 @@ enum JSON_TYPE : byte
     FALSE    /// ditto
 }
 
+private struct DefaultPolicy
+{
+    enum preserveMemberOrder = false;
+}
+private struct PreserveMemberOrderPolicy
+{
+    enum preserveMemberOrder = true;
+}
+
 /**
 JSON value node
 */
-struct JSONValue
+alias JSONValue = JSONValueTemplate!DefaultPolicy;
+
+private struct JSONValueTemplate(alias Policy)
 {
+    alias Value = typeof(this);
+    static if (Policy.preserveMemberOrder)
+    {
+        struct JSONObject
```

Comment:
optional: this is basically an AA with ordered fields; it's re-usable; why not expose it as a generic:
```
struct AAOrdered(K,V){
...
}
```